src/main/java/io/gravitee/resource/oauth2/am/OAuth2AMResource.java

```diff
@@ -229,7 +229,14 @@ public void handle(AsyncResult<HttpClientResponse> asyncResponse) {
                                                         );
                                                     }
                                                 } else {
-                                                    responseHandler.handle(new OAuth2Response(false, buffer.toString()));
+                                                    logger.error(
+                                                        "An error occurs while checking access token. Request ends with status {}: {}",
+                                                        response.statusCode(),
+                                                        buffer.toString()
+                                                    );
+                                                    responseHandler.handle(
+                                                        new OAuth2Response(new Throwable("An error occurs while checking access token"))
```

Comment:
That the limit I pointed out yesterday, I would prefer having an explicit exception here than a generic Throwable.